tmc-langs-r/src/main/java/fi/helsinki/cs/tmc/langs/r/RPlugin.java

```diff
@@ -0,0 +1,158 @@
+                new StudentFileAwareUnzipper());
+    }
+
+    /*
+    NOTE: Files.exists does not seem to be able to verify the R and
+    testthat folder's existence if they are empty.
+    */
+    @Override
+    public boolean isExerciseTypeCorrect(Path path) {
+        return Files.exists(path.resolve(R_FOLDER_PATH))
+                || Files.exists(path.resolve(TEST_FOLDER_PATH).resolve(TESTTHAT_FOLDER_PATH))
+                || Files.exists(path.resolve(TEST_FOLDER_PATH).resolve(TESTTHAT_FILE_PATH));
+    }
+
+    @Override
+    protected StudentFilePolicy getStudentFilePolicy(Path projectPath) {
+        return new RStudentFilePolicy(projectPath);
+    }
+
+    @Override
+    public String getPluginName() {
+        return "r";
+    }
+
+    @Override
+    public Optional<ExerciseDesc> scanExercise(Path path, String exerciseName) {
+        ProcessRunner runner = new ProcessRunner(this.getAvailablePointsCommand(), path);
+
+        try {
+            runner.call();
+        } catch (Exception e) {
+            log.error(CANNOT_SCAN_EXERCISE_MESSAGE, e);
+        }
+
+        try {
+            ImmutableList<TestDesc> testDescs = new RExerciseDescParser(path).parse();
+            return Optional.of(new ExerciseDesc(exerciseName, testDescs));
+        } catch (IOException e) {
+            log.error(CANNOT_PARSE_EXERCISE_DESCRIPTION_MESSAGE, e);
+        }
+
+        return Optional.absent();
+    }
+
+    @Override
+    public RunResult runTests(Path path) {
+        ProcessRunner runner = new ProcessRunner(getTestCommand(), path);
+
+        try {
+            runner.call();
+        } catch (Exception e) {
+            log.error(CANNOT_RUN_TESTS_MESSAGE, e);
+        }
+
+        try {
+            return new RTestResultParser(path).parse();
+        } catch (IOException e) {
+            log.error(CANNOT_PARSE_TEST_RESULTS_MESSAGE, e);
+        }
+
+        return null;
+    }
+
+    @Override
+    public ValidationResult checkCodeStyle(Path path, Locale messageLocale) {
+        return null;
+    }
+
+    public String[] getTestCommand() {
+        String[] command = new String[]{"Rscript"};
+        String[] args;
+        if (SystemUtils.IS_OS_WINDOWS) {
+            args = new String[]{"-e", "\"library('tmcRtestrunner');run_tests()\""};
+        } else {
+            args = new String[]{"-e", "library(tmcRtestrunner);run_tests()"};
+        }
+        return ArrayUtils.addAll(command, args);
+    }
+
+    public String[] getAvailablePointsCommand() {
+        String[] command = new String[]{"Rscript"};
+        String[] args;
+        if (SystemUtils.IS_OS_WINDOWS) {
+            args = new String[]{"-e", "\"library('tmcRtestrunner');run_available_points()\""};
+        } else {
+            args = new String[]{"-e", "library(tmcRtestrunner);run_available_points()"};
+        }
+        return ArrayUtils.addAll(command, args);
+    }
+
+    @Override
+    public void clean(Path path) {
+        // TO DO
```

Comment:
I'd like to see either a comment that effectively says "This is a no-op, and that's fine" or alternatively have this throw an `UnsupportedOperationException` to indicate that it should not be called.

Also document what this method **should** do, in case it gets forgotten and somebody stumbles upon it in this state a few years down the line.